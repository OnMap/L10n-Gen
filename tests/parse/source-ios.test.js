const { collectiOSKeys, localizationiOSFiles } = require('../../src/parse/source-ios');

describe('iOS source parse', () => {
  it('find all iOS oriented files to parse', () => {
    const result = localizationiOSFiles('./tests/__mocks__');
    expect(result).toMatchSnapshot();
  });

  it('find only iOS oriented files in folder', () => {
    const noResult = localizationiOSFiles('./tests/parse');
    expect(noResult).toEqual([]);
  });

  it('collect keys in iOS oriented files', async () => {
    const files = localizationiOSFiles('./tests/__mocks__');
    const result = await collectiOSKeys(files);
    expect(result).toMatchSnapshot();
  });
});
