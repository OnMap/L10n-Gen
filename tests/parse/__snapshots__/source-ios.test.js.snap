// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iOS source parse collect keys in iOS oriented files 1`] = `
Array [
  "Search.Map",
  "Search.List",
  "Search.Filters",
  "Map.MapArea",
  "Map.Loading",
  "Map.NoResults",
  "Main.SearchBar",
  "Main.TextView",
  "Main.TextField",
  "Main.ChangeLanguageButton",
  "Main.StoryboardLabel",
  "Main.SegmentedControl",
  "Main.TitleNavigationItem",
]
`;

exports[`iOS source parse find all iOS oriented files to parse 1`] = `
Array [
  "tests/__mocks__/files/ios/source/Localizable.swift",
  "tests/__mocks__/files/ios/source/Main.storyboard",
]
`;
